File: app/src/app/shared/models/workload.ts

export enum Implementation {
    REPO = "Java - Spring Data repositiory",
    SPRING = "Java - Spring Data Mongo",
    DRIVER = "MongoDB java sync driver",
    NODEJS = "node.js",
    PYTHON = "Python"
}
export enum WorkloadType {
    READ = "Read",
    WRITE = "Write"
}
export enum OperationType {
    INSERT = "Insert",
    UPDATE = "Update",
    DELETE = "Delete",
    REPLACE = "Replace"
}
export enum WriteConcern {
    ACKNOWLEDGED = "Acknowledge",
    W1 = "1",
    W2 = "2",
    W3 = "3",
    UNACKNOWLEDGED = "Unacknowledge",
    JOURNALED = "Journaled",
    MAJORITY = "Majority"
}
export enum Converter {
    SPRING = "Spring converter",
    MONGODB = "MongoDB Codec",
}
export class Workload {
    impl!: Implementation;
    type!: WorkloadType;
    coll!: string;
    entity!: string;
    schema!: string;
    converter!: Converter;
    //TODO:Read options?

    //Write Option
    opType!: OperationType;
    numWorkers!: number;
    qty!: number;
    w!: WriteConcern;
    bulk!: boolean;

    ids: string[] = [];
    query!: string
}

File: java-api/src/main/java/com/mongodb/javabasic/model/Workload.java
package com.mongodb.javabasic.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Workload {
    public enum Implementation {
        REPO, SPRING, DRIVER
    }

    public enum Type {
        READ, WRITE
    }

    public enum OperationType {
        INSERT, UPDATE, DELETE, REPLACE
    }

    public enum WriteConcern {
        ACKNOWLEDGED, W1, W2, W3, UNACKNOWLEDGED, JOURNALED, MAJORITY
    }

    public enum Converter {
        SPRING,
        MONGODB
    }

    @JsonProperty("impl")
    private Implementation implementation;
    private Type type;
    @JsonProperty("docSize")
    private int documentSize;
    private String schema;
    private Converter converter;
    // TODO:Read options?

    // Write Option
    @JsonProperty("opType")
    private OperationType operationType;
    @JsonProperty("numWorkers")
    private int noOfWorkers;
    @JsonProperty("qty")
    private int quantity;
    @JsonProperty("w")
    private WriteConcern writeConcern;
    private boolean bulk;
    private String query;
    private List<String> ids;
}
